### ISSUE TITLE
Container HTTPS doesn't work

### ISSUE THREAD

**Issue by tomlawesome** [shlinkio/shlink#1231]
#### How Shlink is set-up

* Shlink Version: 2.9.2
* PHP Version: (whatever container uses)
* How do you serve Shlink: Self-hosted Docker image
* Database engine used: PostgreSQL

#### Summary

Container responds as expected at http://localip:8080/rest/v2/health with a healthy response. If I switch to http**s**://localip:8080/rest/v2/health, then it doesn't work, despite having the following variables set in my compose file: 

```
- USE_HTTPS=true
- SHORT_DOMAIN_SCHEMA=https
```

When pointing short domain at the docker container (via traefik), I get a 'This domain isn't redirecting properly' error, but nothing shows as erroring in the browser's debugger. 

I host lots of other services and traefik otherwise generally works? 

The shlink web-client also won't connect to the container at a local (or docker) address.

**Comment by acelaya** [comment]
Hhhmm, I think `USE_HTTPS` might be missleading.

That env var does not enable https transparently, it's still up to you to enable it via a reverse proxy in front of Shlink.

The only purpose of this env var is to **inform** Shlink that you are publicly using https, so that it can properly build URLs on API responses.

This env var was introduced in Shlink 2.9 as a replacement of `SHORT_DOMAIN_SCHEMA`, which possible values were `http` or `https`.

In short, Shlink will always respond to http only, and this env var clearly needs to be renamed.

**Comment by acelaya** [comment]
More info in [the docs](https://shlink.io/documentation/environment-variables/)

![image](https://user-images.githubusercontent.com/2719332/141658997-ddca2a4b-1815-42b1-a1cd-faad637df152.png)

**Comment by tomlawesome** [comment]
> 
> 
> Hhhmm, I think `USE_HTTPS` might be missleading.
> 
> That env var does not enable https transparently, it's still up to you to enable it via a reverse proxy in front of Shlink.
> 
> The only purpose of this env var is to **inform** Shlink that you are publicly using https, so that it can properly build URLs on API responses.
> 
> This env var was introduced in Shlink 2.9 as a replacement of `SHORT_DOMAIN_SCHEMA`, which possible values were `http` or `https`.
> 
> In short, Shlink will always respond to http only, and this env var clearly needs to be renamed.

Ok thanks, that definitely clears that up -- I did read that part of the docs, but misunderstood. 

Does shlink webclient care if the SSL certs are different? 

So.. if I connect to shlink web-client at shlink.domain.net and the backend as a default short URL of https://sho.rt using different cert, it should work ok?

**Comment by acelaya** [comment]
Yep, there should be no problem.

For example, I always use the web client at https://app.shlink.io, and I serve my own Shlink behind https://acel.me
